core/src/main/java/org/elasticsearch/index/query/BoolQueryBuilder.java

```diff
@@ -207,4 +323,96 @@ private void doXArrayContent(String field, List<QueryBuilder> clauses, XContentB
+
+        booleanQuery.setBoost(this.boost);
+        Queries.applyMinimumShouldMatch(booleanQuery, this.minimumShouldMatch);
+        Query query = this.adjustPureNegative ? fixNegativeQueryIfNeeded(booleanQuery) : booleanQuery;
+        if (this.queryName != null) {
+            parseContext.addNamedQuery(this.queryName, query);
+        }
+        return query;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        // nothing to validate, clauses are optional, see hasClauses(), other parameters have defaults
+        return null;
+    }
+
+    private void addBooleanClauses(QueryParseContext parseContext, BooleanQuery booleanQuery, List<QueryBuilder> clauses, Occur occurs)
+            throws IOException {
+        for (QueryBuilder query : clauses) {
+            Query luceneQuery = query.toQuery(parseContext);
+            if (luceneQuery != null) {
+                booleanQuery.add(new BooleanClause(luceneQuery, occurs));
+            }
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(boost, adjustPureNegative, disableCoord,
+                minimumShouldMatch, queryName, mustClauses, shouldClauses, mustNotClauses, filterClauses);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        BoolQueryBuilder other = (BoolQueryBuilder) obj;
+        return Objects.equals(boost, other.boost) &&
+                Objects.equals(adjustPureNegative, other.adjustPureNegative) &&
+                Objects.equals(disableCoord, other.disableCoord) &&
+                Objects.equals(minimumShouldMatch, other.minimumShouldMatch) &&
+                Objects.equals(queryName, other.queryName) &&
+                Objects.equals(mustClauses, other.mustClauses) &&
+                Objects.equals(shouldClauses, other.shouldClauses) &&
+                Objects.equals(mustNotClauses, other.mustNotClauses) &&
+                Objects.equals(filterClauses, other.filterClauses);
+    }
+
+    @Override
+    public BoolQueryBuilder readFrom(StreamInput in) throws IOException {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        boolQueryBuilder.must(QueryBuilder.readQueryBuilderListFrom(in));
+        boolQueryBuilder.mustNot(QueryBuilder.readQueryBuilderListFrom(in));
+        boolQueryBuilder.should(QueryBuilder.readQueryBuilderListFrom(in));
+        boolQueryBuilder.filter(QueryBuilder.readQueryBuilderListFrom(in));
+        boolQueryBuilder.boost = in.readFloat();
+        boolQueryBuilder.queryName = in.readOptionalString();
+        boolQueryBuilder.adjustPureNegative = in.readBoolean();
+        boolQueryBuilder.disableCoord = in.readBoolean();
+        boolQueryBuilder.minimumShouldMatch = in.readOptionalString();
+        return boolQueryBuilder;
```

Comment:
here too, similar to my doubts around printing out default vaues. In general, shall we have some discussion on whehter we want to always serialize everything. Sounds good to me, I just wonder if we should look into serializing less fields given that most of the times everything will be default and we could save bandwidth? Or maybe this is anyways not a problem at all?